src/Benchmarks/Middleware/SignalRMiddleware.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Benchmarks.Configuration;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Benchmarks.Middleware
+{
+    public static class SignalRMiddlewareExtensions
+    {
+        public static IApplicationBuilder UseSignalRMiddleware(this IApplicationBuilder builder)
+        {
+            return builder.UseSignalR(route =>
+            {
+                route.MapHub<EchoHub>(Scenarios.GetPath(s => s.SignalRBroadcast) + "/default");
+            });
+        }
+    }
+
+    public class EchoHub : Hub
+    {
+        public override Task OnConnectedAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        public override Task OnDisconnectedAsync(Exception exception)
+        {
+            return Task.CompletedTask;
+        }
+
+        public async Task Echo(int duration)
+        {
+            try
+            {
+                var t = new CancellationTokenSource();
+                t.CancelAfter(TimeSpan.FromSeconds(duration));
+                while (!t.IsCancellationRequested)
```

Comment:
Wait on client disconnected as well?

Reply:
Ok, I'll keep track of how many connections are on the server and when it hits 0 exit the loop

Reply:
You don't have to. You can just make sure you're not sending after this client is gone by checking this token and the aborted token.